app/code/core/Mage/Adminhtml/controllers/Catalog/ProductController.php

```diff
@@ -559,6 +559,7 @@ public function validateAction()
             $response->setMessage($this->getLayout()->getMessagesBlock()->getGroupedHtml());
         }
 
+        $this->getResponse()->setHeader('Content-type', 'application/json', true);
         $this->getResponse()->setBody($response->toJson());
```

Comment:
maybe this could also use the setBodyJson?

Reply:
How did you review it so fast, haha. You're right, I missed that one.